### PR TITLE
Feature/5 remove xmlrpc dependency from api

### LINKED ISSUE #5: Remove XmlRpcException from API

### PR DESCRIPTION
**Pull request by jakobbraun**
closes #5

### CODE REVIEW

File: src/test/java/com/exasol/ExaOperationInterface.java [hunk 1]
```diff
@@ -33,34 +33,56 @@ public ExaOperationInterface(final String exasolIpAddress, final String adminUse
...
     public void createAndUploadJdbcDriver(final String name, final String jdbcMainClass, final String prefix,
-            final boolean disableSecurityManager, final File driverJar) throws XmlRpcException, IOException {
-        final String driverId = addJdbcDriver(name, jdbcMainClass, prefix, disableSecurityManager);
-        uploadJdbcDriver(driverId, driverJar);
+            final boolean disableSecurityManager, final File driverJar) throws IOException {
+        try {
+            final String driverId = addJdbcDriver(name, jdbcMainClass, prefix, disableSecurityManager);
+            uploadJdbcDriver(driverId, driverJar);
+        } catch (final XmlRpcException exception) {
+            throw new IllegalStateException("Failed create or upload JDBC driver. Cause: " + exception.getMessage(),
```
Comment (morazow) [inline src/test/java/com/exasol/ExaOperationInterface.java:61]:
```suggestion
            throw new IllegalStateException("Failed to create or upload JDBC driver. Cause: " + exception.getMessage(),
```

Comment (morazow) [reply src/test/java/com/exasol/ExaOperationInterface.java:61]: Please also add to other error message to make it consistent with above ones.

File: src/test/java/com/exasol/ExaOperationInterface.java [hunk 2]
```diff
@@ -33,34 +33,56 @@ public ExaOperationInterface(final String exasolIpAddress, final String adminUse
...
+                    exception);
+        }
     }
 
     private String addJdbcDriver(final String name, final String jdbcMainClass, final String prefix,
             final boolean disableSecurityManager) throws XmlRpcException {
```
Comment (morazow) [inline src/test/java/com/exasol/ExaOperationInterface.java:67]: Remove throws XmlRpcException.

File: src/test/java/com/exasol/ExaOperationInterface.java [hunk 3]
```diff
@@ -33,34 +33,56 @@ public ExaOperationInterface(final String exasolIpAddress, final String adminUse
...
 
     public void createAndUploadJdbcDriver(final String name, final String jdbcMainClass, final String prefix,
-            final boolean disableSecurityManager, final File driverJar) throws XmlRpcException, IOException {
-        final String driverId = addJdbcDriver(name, jdbcMainClass, prefix, disableSecurityManager);
-        uploadJdbcDriver(driverId, driverJar);
+            final boolean disableSecurityManager, final File driverJar) throws IOException {
+        try {
+            final String driverId = addJdbcDriver(name, jdbcMainClass, prefix, disableSecurityManager);
+            uploadJdbcDriver(driverId, driverJar);
+        } catch (final XmlRpcException exception) {
```
Comment (morazow) [inline src/test/java/com/exasol/ExaOperationInterface.java:60]: Catch IOException and change error message accordingly.

File: src/test/java/com/exasol/ExaOperationInterface.java [hunk 4]
```diff
@@ -33,34 +33,56 @@ public ExaOperationInterface(final String exasolIpAddress, final String adminUse
...
+        }
     }
 
-    private void uploadJdbcDriver(final String jdbcDriverId, final File file) throws IOException, XmlRpcException {
+    private void uploadJdbcDriver(final String jdbcDriverId, final File file) throws IOException {
         try (final InputStream inputStream = new FileInputStream(file)) {
             final byte[] driverBytes = IOUtils.toByteArray(inputStream);
             this.client.execute(jdbcDriverId + ".uploadFile", new Object[] { driverBytes, file.getName() });
         } catch (final XmlRpcClientException exception) {
             // This exception happens always. I don't know why but it still works...
```
Comment (morazow) [inline src/test/java/com/exasol/ExaOperationInterface.java:82]: Does exception still happens and this comment still valid?

**Review by morazow — CHANGES_REQUESTED** [review]

File: src/test/java/com/exasol/ExaOperationInterface.java (hunk 4 above)
Comment (jakobbraun) [reply src/test/java/com/exasol/ExaOperationInterface.java:82]: I think so ;-)

**Review by morazow — APPROVED** [review]

**Pull request merged**